tests/src/test-agent.ts

```diff
@@ -186,15 +186,68 @@ test("stats are acquired", async (t) => {
   expectResponse(t, nats, "drone.1.spawn", request, true)
   await sleep(100)
 
-  const statsStatusSubsription = 
+  const statsStatusSubsription =
     new NatsMessageIterator<unknown>(
-      nats.subscribe(`backend.${backendId}.stats`, { timeout : 1000000 })
-  )
+      nats.subscribe(`backend.${backendId}.stats`, { timeout: 1000000 })
+    )
   const statsMessage = await statsStatusSubsription.next()
-  const stats= statsMessage[0]
+  const stats = statsMessage[0]
   t.assert(stats["cpu_use_percent"] > 0 && stats["mem_use_percent"] > 0)
 })
 
+test("stats are killed after container dies", async (t) => {
+  const backendId = generateId()
+  const natsPort = await t.context.docker.runNats()
+  await sleep(1000)
+  const nats = await connect({ port: natsPort, token: "mytoken" })
+  t.context.runner.runAgent(natsPort)
+
+  await expectMessage(t, nats, "drone.register", {
+    cluster: "mydomain.test",
+    ip: "123.12.1.123"
+  }, {
+    Success: {
+      drone_id: 1,
+    }
+  })
+
+  await sleep(100)
+
+  const request: SpawnRequest = {
+    image: TEST_IMAGE,
+    backend_id: backendId,
+    max_idle_secs: 10,
+    env: {
+      PORT: "8080",
+    },
+    metadata: {},
+  }
+
+  expectResponse(t, nats, "drone.1.spawn", request, true)
+
+  await sleep(100)
+
+  const statsStatusSubsription =
+    new NatsMessageIterator<unknown>(
+      nats.subscribe(`backend.${backendId}.stats`, { timeout: 1000000 })
+    )
+
+  //TODO: figure out a way to write a test such that
```

Comment:
Does this TODO not describe what's implemented below? We can probably just remove it, or move it into a linear issue if it's something you want to do in the next few cycles.

Reply:
yea, what I'd like is a nice test.eventuallyEnds(nats.subscription, TIMEOUT), but I'll stick that into linear backlog.